display_list/skia/dl_sk_paint_dispatcher.h

```diff
@@ -37,7 +37,19 @@ class DlSkPaintDispatchHelper : public virtual DlOpReceiver {
   void setMaskFilter(const DlMaskFilter* filter) override;
   void setImageFilter(const DlImageFilter* filter) override;
 
-  const SkPaint& paint() { return paint_; }
+  const SkPaint& paint() {
+    // On the Impeller backend, we will only support dithering of *gradients*,
+    // and it will be enabled by default (without the option to disable it).
+    // Until Skia support is completely removed, we only want to respect the
+    // dither flag for gradients (otherwise it will also apply to, for
+    // example, images, which is not supported in Impeller).
+    //
+    // See https://github.com/flutter/flutter/issues/112498.
+    if (!color_source_gradient_) {
+      paint_.setDither(false);
```

Comment:
Should this hinge on setDither(true) having been called?

So, 2 flags
- is_gradient, set from setColorSource as now
- is_dither, set from setDither (in lieu of setting sk_paint.setDither())

In this method, just paint_.setDither(is_gradient && is_dither);

Reply:
Keep in mind that we might inherit a color source and/or a dither setting across multiple rendering calls. Those methods are only called when the associated attribute *changes*, they aren't called on every single rendering op.

So, you have to track the "last set values" for both and make your sk_paint dithering decision based on those last values, just on "a recently seen value" for one of them.

Reply:
I think where this will likely fail is
- dither true
- color_source non-gradient
- render something
- color_source gradient
- render something - your code will have the flag set to false in this case I think...?

Reply:
Done, and done, thanks for the suggestion and added the last case as a test!